creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -377,37 +388,73 @@ class SnapbackSM {
-    let currentNodeHasExistingUserState = true
+    let newReplicaNodesSet = new Set()
     let i = 0
-    let newSecondary
-    while (currentNodeHasExistingUserState && i++ < MAX_SELECT_NEW_REPLICA_SET_ATTEMPTS) {
-      // Make sure current replica set will not be chosen as potential nodes for the new replica set
-      const healthyPotentialPeers = healthyNodes.filter(node => node !== primary && node !== secondary1 && node !== secondary2)
-
+    let issueReconfig = false
+    let newPrimary, newSecondary1, newSecondary2
+    while (newReplicaNodesSet.size !== NUMBER_OF_REPLICA_SET_NODES && i++ < MAX_SELECT_NEW_REPLICA_SET_ATTEMPTS) {
       // Select a random node that is not from the current replica set
-      newSecondary = healthyPotentialPeers[Math.floor(Math.random() * healthyPotentialPeers.length)]
+      let randomHealthyNode = healthyNodes[Math.floor(Math.random() * healthyNodes.length)]
+
+      // If node is already present in set, keep finding a unique healthy node
+      if (newReplicaNodesSet.has(randomHealthyNode)) continue
 
       // Check to make sure that the newly selected secondary does not have existing user state
       try {
-        const clockValue = await CreatorNode.getClockValue(newSecondary, wallet)
-        if (clockValue === -1) currentNodeHasExistingUserState = false
+        const clockValue = await CreatorNode.getClockValue(randomHealthyNode, wallet)
+        if (clockValue === -1) {
+          newReplicaNodesSet.add(randomHealthyNode)
+        }
       } catch (e) {
         // Something went wrong in checking clock value. Reselect another secondary.
-        this.logError(e.message)
+        this.logError(`[determineNewReplicaSet] ${e.message}`)
       }
     }
 
-    if (currentNodeHasExistingUserState) throw new Error(`Unable to select new secondary given current healthy Content Nodes=[${healthyNodes}]`)
-
-    const currentHealthySecondary = unhealthyReplicasSet.has(secondary1) ? secondary2 : secondary1
+    let newReplicaNodes = Array.from(newReplicaNodesSet)
+    // We need a minimum of `NUMBER_OF_REPLICA_SET_NODES` to perform the reconfig op. In the case that
+    // the max number replica set attempts were made, push null
+    while (newReplicaNodes.length < NUMBER_OF_REPLICA_SET_NODES) {
+      newReplicaNodes.push(null)
+    }
 
-    // NOTE: For v0 implementation, `newSecondary2` will be the newly selected secondary, whilst the other nodes will be the
-    // original replica set primary and secondary1
-    return {
-      newPrimary: primary,
-      newSecondary1: currentHealthySecondary,
-      newSecondary2: newSecondary
+    // Note: for v0, only issue reconfigs if only 1 secondary is unhealthy. Else, just log out the tentative
+    // new replica set to be assigned.
+
+    if (unhealthyReplicasSet.size === 1) {
+      if (unhealthyReplicasSet.has(primary)) {
+        // If the primary is unhealthy, select the higher clock value of the two secondaries as the new primary,
```

Comment:
this case is the most dangerous one, esp in the case that the primary has clock value higher than both secondaries
something to think about, we'll need to flesh out this logic more

Reply:
can discuss via video chat later